consensus/polybft/consensus_runtime.go

```diff
@@ -417,81 +440,68 @@ func (c *consensusRuntime) FSM() error {
 }
 
 // restartEpoch resets the previously run epoch and moves to the next one
-func (c *consensusRuntime) restartEpoch(header *types.Header) error {
+// returns *epochMetadata different from nil if the lastEpoch is not the current one and everything was successful
+func (c *consensusRuntime) restartEpoch(header *types.Header, lastEpoch *epochMetadata) (*epochMetadata, error) {
```

Comment:
I would say we don't need this `lastEpoch` parameter, since it's already in the runtime object, and the function is under lock.

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1047/commits/9ad952c1a99a2bacf73e1e6a2146d02f22021ec2